{
    "active_lock_reason": null,
    "assignee": null,
    "assignees": [],
    "author_association": "CONTRIBUTOR",
    "body": "### Problem\n\nWhen I try to run a Pyhton expression, I get that there is not python provider:\r\n\r\n```\r\n:lua vim.fn.py3eval(\"2+2\")                                                                                                               \r\nE5108: Error executing lua Vim:E319: No \"python3\" provider found. Run \":checkhealth provider\"                                            \r\nstack traceback:                                                                                                                         \r\n        [C]: in function 'py3eval'                                                                                                       \r\n        [string \":lua\"]:1: in main chunk\r\n```\r\n\r\nHowever there is.\r\n\r\nHere is the output of `:checkhealth provider.python`:\r\n```\r\nprovider.python: require(\"provider.python.health\").check()                                                                           \r\n                                                                                                                                        \r\n Python 3 provider (optional)                                                                                                         \r\n - pyenv: Path: /usr/share/pyenv/libexec/pyenv                                                                                        \r\n - pyenv: Root: /home/USERNAME/.pyenv                                                                                                 \r\n - Using: g:python3_host_prog = \"/home/USERNAME/.pyenv/versions/neovim/bin/python3\"                                                   \r\n - Executable: /home/USERNAME/.pyenv/versions/neovim/bin/python3                                                                      \r\n - Python version: 3.11.7                                                                                                             \r\n - pynvim version: 0.5.0                                                                                                              \r\n - OK Latest pynvim is installed.                                                                                                     \r\n                                                                                                                                       \r\n Python virtualenv                                                                                                                    \r\n - OK no $VIRTUAL_ENV                                               \r\n```\r\n\r\n\r\n\r\n\n\n### Steps to reproduce\n\n1. Create a virtual environment with pyenv: `pyenv virtualenv 3.12 neovim`\r\n2. Activate it `pyenv local neovim`\r\n3. Install pynvim on it: `python -m pip install pynvim`\r\n4. Set the virtual environment as python provider in init.lua:\r\n```\r\n-- WARNING: A python 3 provider is needed for zotcite\r\nvim.g.loaded_python3_provider = 1\r\n-- WARNING: Path must be set to a `python3` executable file,\r\n--          NOT to a `python` executable file.\r\n--          That won't work.\r\nvim.g.python3_host_prog = MYHOME .. \"/.pyenv/versions/neovim/bin/python3\"\r\n```\r\n5. Run `:lua vim.fn.py3eval(\"2+2\")`\n\n### Expected behavior\n\nneovim should not error out as it says the provider is present\n\n### Neovim version (nvim -v)\n\nv0.10.1\n\n### Vim (not Nvim) behaves the same?\n\nno\n\n### Operating system/version\n\nArch Linux\n\n### Terminal name/version\n\nkitty 0.36.1\n\n### $TERM environment variable\n\nxterm-kitty\n\n### Installation\n\nArch Linux official package",
    "closed_at": null,
    "closed_by": null,
    "comment_data": [
        {
            "author_association": "MEMBER",
            "body": "Does this happen with `nvim --clean` ? \r\n\r\n\r\nCan you try the development version? The [Releases](https://github.com/neovim/neovim/releases/tag/nightly) page has pre-built archives for Linux/Windows/macOS. Check `:version` to confirm the version.",
            "created_at": "2024-09-04T18:26:21Z",
            "html_url": "https://github.com/neovim/neovim/issues/30263#issuecomment-2329715814",
            "id": 2329715814,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/30263",
            "node_id": "IC_kwDOAPphoM6K3KRm",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2329715814/reactions"
            },
            "updated_at": "2024-09-04T18:26:59Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2329715814",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/1359421?v=4",
                "events_url": "https://api.github.com/users/justinmk/events{/privacy}",
                "followers_url": "https://api.github.com/users/justinmk/followers",
                "following_url": "https://api.github.com/users/justinmk/following{/other_user}",
                "gists_url": "https://api.github.com/users/justinmk/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/justinmk",
                "id": 1359421,
                "login": "justinmk",
                "node_id": "MDQ6VXNlcjEzNTk0MjE=",
                "organizations_url": "https://api.github.com/users/justinmk/orgs",
                "received_events_url": "https://api.github.com/users/justinmk/received_events",
                "repos_url": "https://api.github.com/users/justinmk/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/justinmk/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/justinmk/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/justinmk",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "CONTRIBUTOR",
            "body": "> Does this happen with nvim --clean ?\r\n\r\nNo, with `nvim--clean` it works\r\n\r\n> Can you try the development version? The [Releases](https://github.com/neovim/neovim/releases/tag/nightly) page has pre-built archives for Linux/Windows/macOS. Check :version to confirm the version.\r\n\r\nSame thing:\r\n```                                                                                                                                     \r\n:version                                                                                                                                                                                                                                           \r\nNVIM v0.11.0-dev-711+g7b7c95dac                                                                                                          \r\nBuild type: RelWithDebInfo                                                                                                               \r\nLuaJIT 2.1.1724512491                                                                                                                    \r\nRun \":verbose version\" for more info\r\n\r\n:lua vim.fn.py3eval(\"1+1\")                                                                                                               \r\nE5108: Error executing lua Vim:E319: No \"python3\" provider found. Run \":checkhealth provider\"                                            \r\nstack traceback:                                                                                                                         \r\n        [C]: in function 'py3eval'                                                                                                       \r\n        [string \":lua\"]:1: in main chunk \r\n\r\n:py3 2+2                                                                                                                                 \r\nE319: No \"python3\" provider found. Run \":checkhealth provider\" \r\n\r\n:checkhealth provider\r\nPython 3 provider (optional) ~\r\n- pyenv: Path: /usr/share/pyenv/libexec/pyenv\r\n- pyenv: Root: /home/raffaele/.pyenv\r\n- `g:python3_host_prog` is not set. Searching for python3 in the environment.\r\n- Executable: /usr/bin/python3\r\n- Python version: 3.12.5\r\n- pynvim version: 0.5.0\r\n- OK Latest pynvim is installed.\r\n\r\nPython virtualenv ~\r\n- OK no $VIRTUAL_ENV\r\n\r\n```",
            "created_at": "2024-09-04T22:54:26Z",
            "html_url": "https://github.com/neovim/neovim/issues/30263#issuecomment-2330286666",
            "id": 2330286666,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/30263",
            "node_id": "IC_kwDOAPphoM6K5VpK",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2330286666/reactions"
            },
            "updated_at": "2024-09-04T23:01:37Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2330286666",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/54762742?v=4",
                "events_url": "https://api.github.com/users/raffaem/events{/privacy}",
                "followers_url": "https://api.github.com/users/raffaem/followers",
                "following_url": "https://api.github.com/users/raffaem/following{/other_user}",
                "gists_url": "https://api.github.com/users/raffaem/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/raffaem",
                "id": 54762742,
                "login": "raffaem",
                "node_id": "MDQ6VXNlcjU0NzYyNzQy",
                "organizations_url": "https://api.github.com/users/raffaem/orgs",
                "received_events_url": "https://api.github.com/users/raffaem/received_events",
                "repos_url": "https://api.github.com/users/raffaem/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/raffaem/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/raffaem/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/raffaem",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "Why are you setting `vim.g.loaded_python3_provider = 1`? If you want to load the Python3 provider, you should not set the variable before it's loaded.",
            "created_at": "2024-09-05T00:37:01Z",
            "html_url": "https://github.com/neovim/neovim/issues/30263#issuecomment-2330379381",
            "id": 2330379381,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/30263",
            "node_id": "IC_kwDOAPphoM6K5sR1",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 1,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 1,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2330379381/reactions"
            },
            "updated_at": "2024-09-05T00:38:58Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2330379381",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/35768171?v=4",
                "events_url": "https://api.github.com/users/zeertzjq/events{/privacy}",
                "followers_url": "https://api.github.com/users/zeertzjq/followers",
                "following_url": "https://api.github.com/users/zeertzjq/following{/other_user}",
                "gists_url": "https://api.github.com/users/zeertzjq/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/zeertzjq",
                "id": 35768171,
                "login": "zeertzjq",
                "node_id": "MDQ6VXNlcjM1NzY4MTcx",
                "organizations_url": "https://api.github.com/users/zeertzjq/orgs",
                "received_events_url": "https://api.github.com/users/zeertzjq/received_events",
                "repos_url": "https://api.github.com/users/zeertzjq/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/zeertzjq/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/zeertzjq/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/zeertzjq",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "CONTRIBUTOR",
            "body": "> Why are you setting `vim.g.loaded_python3_provider = 1`? If you want to load the Python3 provider, you should not set the variable before it's loaded.\r\n\r\nWell to parallel\r\n```\r\nvim.g.loaded_ruby_provider = 0\r\nvim.g.loaded_node_provider = 0\r\nvim.g.loaded_perl_provider = 0\r\n```\r\n\r\nThat I have before in the init file.\r\n\r\nRemoving `vim.g.loaded_python3_provider = 1` fixes the issue.\r\n\r\nI guess this bug report should stay as \":checkhealth should not tell me that everything is ok\" and/or \"the documentation should make it clear that you don't want to set this to 1, but only to 0\"?",
            "created_at": "2024-09-05T06:34:54Z",
            "html_url": "https://github.com/neovim/neovim/issues/30263#issuecomment-2330713311",
            "id": 2330713311,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/30263",
            "node_id": "IC_kwDOAPphoM6K69zf",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2330713311/reactions"
            },
            "updated_at": "2024-09-05T06:34:54Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/2330713311",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/54762742?v=4",
                "events_url": "https://api.github.com/users/raffaem/events{/privacy}",
                "followers_url": "https://api.github.com/users/raffaem/followers",
                "following_url": "https://api.github.com/users/raffaem/following{/other_user}",
                "gists_url": "https://api.github.com/users/raffaem/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/raffaem",
                "id": 54762742,
                "login": "raffaem",
                "node_id": "MDQ6VXNlcjU0NzYyNzQy",
                "organizations_url": "https://api.github.com/users/raffaem/orgs",
                "received_events_url": "https://api.github.com/users/raffaem/received_events",
                "repos_url": "https://api.github.com/users/raffaem/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/raffaem/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/raffaem/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/raffaem",
                "user_view_type": "public"
            }
        }
    ],
    "comments": 4,
    "comments_url": "https://api.github.com/repos/neovim/neovim/issues/30263/comments",
    "created_at": "2024-09-04T16:03:05Z",
    "events_url": "https://api.github.com/repos/neovim/neovim/issues/30263/events",
    "html_url": "https://github.com/neovim/neovim/issues/30263",
    "id": 2505767008,
    "labels": [
        {
            "color": "f9d0c4",
            "default": true,
            "description": "issues reporting wrong behavior",
            "id": 77997474,
            "name": "bug",
            "node_id": "MDU6TGFiZWw3Nzk5NzQ3NA==",
            "url": "https://api.github.com/repos/neovim/neovim/labels/bug"
        },
        {
            "color": "c5def5",
            "default": false,
            "description": null,
            "id": 144672485,
            "name": "provider",
            "node_id": "MDU6TGFiZWwxNDQ2NzI0ODU=",
            "url": "https://api.github.com/repos/neovim/neovim/labels/provider"
        },
        {
            "color": "BFDADC",
            "default": false,
            "description": "Low-risk, self-contained. Do NOT ask \"can I work on this\", just read CONTRIBUTING.md",
            "id": 407246773,
            "name": "complexity:low",
            "node_id": "MDU6TGFiZWw0MDcyNDY3NzM=",
            "url": "https://api.github.com/repos/neovim/neovim/labels/complexity:low"
        },
        {
            "color": "c5def5",
            "default": false,
            "description": "",
            "id": 7015246296,
            "name": "checkhealth",
            "node_id": "LA_kwDOAPphoM8AAAABoiQp2A",
            "url": "https://api.github.com/repos/neovim/neovim/labels/checkhealth"
        }
    ],
    "labels_url": "https://api.github.com/repos/neovim/neovim/issues/30263/labels{/name}",
    "locked": false,
    "milestone": {
        "closed_at": null,
        "closed_issues": 686,
        "created_at": "2014-05-10T20:43:04Z",
        "creator": {
            "avatar_url": "https://avatars.githubusercontent.com/u/1359421?v=4",
            "events_url": "https://api.github.com/users/justinmk/events{/privacy}",
            "followers_url": "https://api.github.com/users/justinmk/followers",
            "following_url": "https://api.github.com/users/justinmk/following{/other_user}",
            "gists_url": "https://api.github.com/users/justinmk/gists{/gist_id}",
            "gravatar_id": "",
            "html_url": "https://github.com/justinmk",
            "id": 1359421,
            "login": "justinmk",
            "node_id": "MDQ6VXNlcjEzNTk0MjE=",
            "organizations_url": "https://api.github.com/users/justinmk/orgs",
            "received_events_url": "https://api.github.com/users/justinmk/received_events",
            "repos_url": "https://api.github.com/users/justinmk/repos",
            "site_admin": false,
            "starred_url": "https://api.github.com/users/justinmk/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/justinmk/subscriptions",
            "type": "User",
            "url": "https://api.github.com/users/justinmk",
            "user_view_type": "public"
        },
        "description": "Low priority items. We plan to work on this but don't have a target date.",
        "due_on": null,
        "html_url": "https://github.com/neovim/neovim/milestone/6",
        "id": 655037,
        "labels_url": "https://api.github.com/repos/neovim/neovim/milestones/6/labels",
        "node_id": "MDk6TWlsZXN0b25lNjU1MDM3",
        "number": 6,
        "open_issues": 635,
        "state": "open",
        "title": "backlog",
        "updated_at": "2025-01-21T00:09:37Z",
        "url": "https://api.github.com/repos/neovim/neovim/milestones/6"
    },
    "node_id": "I_kwDOAPphoM6VWvhg",
    "number": 30263,
    "performed_via_github_app": null,
    "reactions": {
        "+1": 0,
        "-1": 0,
        "confused": 0,
        "eyes": 0,
        "heart": 0,
        "hooray": 0,
        "laugh": 0,
        "rocket": 0,
        "total_count": 0,
        "url": "https://api.github.com/repos/neovim/neovim/issues/30263/reactions"
    },
    "repository_url": "https://api.github.com/repos/neovim/neovim",
    "state": "open",
    "state_reason": null,
    "sub_issues_summary": {
        "completed": 0,
        "percent_completed": 0,
        "total": 0
    },
    "timeline_url": "https://api.github.com/repos/neovim/neovim/issues/30263/timeline",
    "title": ":checkhealth should detect if `g:loaded_python3_provider=1` was wrongly set",
    "updated_at": "2025-01-19T10:55:35Z",
    "url": "https://api.github.com/repos/neovim/neovim/issues/30263",
    "user": {
        "avatar_url": "https://avatars.githubusercontent.com/u/54762742?v=4",
        "events_url": "https://api.github.com/users/raffaem/events{/privacy}",
        "followers_url": "https://api.github.com/users/raffaem/followers",
        "following_url": "https://api.github.com/users/raffaem/following{/other_user}",
        "gists_url": "https://api.github.com/users/raffaem/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/raffaem",
        "id": 54762742,
        "login": "raffaem",
        "node_id": "MDQ6VXNlcjU0NzYyNzQy",
        "organizations_url": "https://api.github.com/users/raffaem/orgs",
        "received_events_url": "https://api.github.com/users/raffaem/received_events",
        "repos_url": "https://api.github.com/users/raffaem/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/raffaem/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/raffaem/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/raffaem",
        "user_view_type": "public"
    }
}